lib/Sema/CSFix.h

```diff
@@ -1918,6 +1920,31 @@ class SpecifyKeyPathRootType final : public ConstraintFix {
                                           ConstraintLocator *locator);
 };
 
+/// Diagnose missing unwrap of optional base type on key path application.
+///
+/// \code
+/// func f(_ bar: Bar? , keyPath: KeyPath<Bar, Int>) {
+///   bar[keyPath: keyPath] 
+/// }
+/// \endcode
+class UnwrapOptionalBaseKeyPathApplication final : public ContextualMismatch {
+protected:
+  UnwrapOptionalBaseKeyPathApplication(ConstraintSystem &cs, Type lhs, Type rhs,
+                               ConstraintLocator *locator)
```

Comment:
Nit: Does this require a `clang-format`?

Reply:
humm yeah, I just forgot to run it here ... there is a way we can run `clang-format` only on diff(hear of it somewhere else)? I normally run it block by block with `--lines`

Reply:
there is `git-clang-format` in `llvm-project/clang/tools/clang-format` you can add that to your `PATH` and use right after you have `git add`ed files.

Reply:
Awesome! Thanks @xedin :)